go.mod

```diff
@@ -5,6 +5,7 @@ go 1.13
 require (
 	github.com/cloudflare/cfssl v1.4.1
 	github.com/fsnotify/fsnotify v1.4.9
+	github.com/gogo/protobuf v1.3.2 // indirect
```

Comment:
should this be put in a replace directive ? 

https://github.com/k8snetworkplumbingwg/sriov-network-operator/blob/015d6e055bf0bc27f6d8da74509b3bee433e9613/go.mod#L44

Reply:
I will make the change.
So if we have multiple versions in go.sum, it means all those versions are used directly/indirectly in the repo, not just the tip version?

Reply:
Thats what i understand from: https://github.com/golang/go/wiki/Modules#when-should-i-use-the-replace-directive

Reply:
looking at `go.sum` , i see gogo/protobuf v1.3.0 and 1.3.1 which is not present if you use the replace statement.
see : https://github.com/k8snetworkplumbingwg/sriov-network-operator/blob/015d6e055bf0bc27f6d8da74509b3bee433e9613/go.sum#L310

Reply:
> looking at `go.sum` , i see gogo/protobuf v1.3.0 and 1.3.1 which is not present if you use the replace statement.
> see : https://github.com/k8snetworkplumbingwg/sriov-network-operator/blob/015d6e055bf0bc27f6d8da74509b3bee433e9613/go.sum#L310
> 
> if you delete and recreate go.sum file are these old versions still there ? I assume if they are there, something is using it.

They are still there if I re-create go.sum, but they may not be used: https://github.com/golang/go/wiki/Modules#is-gosum-a-lock-file-why-does-gosum-include-information-for-module-versions-i-am-no-longer-using

Reply:
Minimum selection algorithm ensures 1.3.1 is chosen for version 1 of protobuf. See [here](https://github.com/golang/go/wiki/Modules#version-selection). Its fine if lower versions of v1 is seen in go.sum. It will not be chosen.
Going forward, are we using go get and verify with go list -m all? or replace directive? I'd rather go get but I have no reason why except its what I see others use. We should also run go mod tidy after any action on go.mod.

Reply:
@adrianchiris are you ok with this path - using "go get ..." instead of replace directive to force upgrading components? Or would you rather use replace directive? I am open to hearing your experienced experience on this and knowledge from how other repos do it. I think go get and verify with go list -m all is fine for upgrading.

Reply:
> Minimum selection algorithm ensures 1.3.2 is chosen for version 1 of protobuf. See here. Its fine if lower versions of v1 is seen in go.sum. It will not be chosen.
Going forward, are we using go get and verify with go list -m all? or replace directive? I'd rather go get but I have no reason why except its what I see others use. We should also run go mod tidy after any action on go.mod.

I see, thanks for looking at this Martin!
So it seems that both methods: go get , replace are valid in overriding old module version.
i know the latter works. And asked zenghui to ensure the former indeed works before we merge.

So, TL;DR, if both works i prefer the "go get" as if later on we depend on a module with a newer version of a dependency, we will not break the module by overriding the dependency with a lower version.